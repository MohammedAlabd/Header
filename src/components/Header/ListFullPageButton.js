import React, { useState } from "react";
import ExpandLess from "@material-ui/icons/ExpandLess";
import ExpandMore from "@material-ui/icons/ExpandMore";
import { Button, Menu, MenuItem, Typography } from "@material-ui/core";
import { NavLink } from "react-router-dom";
import { makeStyles } from "@material-ui/core/styles";

const useStyles = makeStyles(() => ({
  textDecoration: {
    textDecoration: "none",
  },
}));

export default function ListFullPage({ page, currentPagePath }) {
  const classes = useStyles();

  const [anchorEl, setAnchorEl] = useState(null);

  const handleOpenMenu = (event) => {
    setAnchorEl(event.currentTarget);
  };

  const handleCloseMenu = () => {
    setAnchorEl(null);
  };

  // this regex used to search for the page name in the url
  const listName = new RegExp(`/${page.url}`, "i");
  const isListSelected = currentPagePath.search(listName) === 0;

  return (
    <>
      <Button
        data-testid="openMenuButton"
        color={isListSelected ? "primary" : "secondary"}
        onClick={handleOpenMenu}
      >
        <Typography
          color={isListSelected ? "primary" : "secondary"}
          variant="subtitle1"
        >
          {page.name}
        </Typography>
        {Boolean(anchorEl) ? <ExpandLess /> : <ExpandMore />}
      </Button>
      <Menu
        data-testid="menu"
        anchorEl={anchorEl}
        keepMounted
        open={Boolean(anchorEl)}
        onClose={handleCloseMenu}
      >
        {page.list.map((listItem) => {
          const isPageSelected =
            currentPagePath === `/${page.url}/${listItem.url}`;

          return (
            <NavLink
              onClick={handleCloseMenu}
              to={`/${page.url}/${listItem.url}`}
              className={classes.textDecoration}
              key={listItem.url}
            >
              <MenuItem>
                <Typography
                  variant="subtitle1"
                  color={isPageSelected ? "primary" : "secondary"}
                >
                  {listItem.name.toLocaleUpperCase()}
                </Typography>
              </MenuItem>
            </NavLink>
          );
        })}
      </Menu>
    </>
  );
}
