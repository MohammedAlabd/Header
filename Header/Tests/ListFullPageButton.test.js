import React from "react";
import ListFullPageButton from "../ListFullPageButton";
import "@testing-library/jest-dom";
import { render, fireEvent } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

describe("ListFullPageButton Test", () => {
  const page = {
    name: "Page",
    url: "page-url", //to make the color of the button primary
    list: [
      {
        name: "subPage1",
        url: "subPage1-url", //to make the color of the button primary
      },
    ],
  };
  let tree;
  beforeEach(() => {
    tree = render(
      <MemoryRouter>
        <ListFullPageButton
          page={page}
          currentPagePath={"/page-url/subPage1-url"} //to make the color of the button primary
        />
      </MemoryRouter>
    );
  });

  it("matches snapshot", () => {
    expect(tree).toMatchSnapshot();
  });

  it("open the correct menu", () => {
    let btn = tree.getByTestId("openMenuButton");
    fireEvent.click(btn);
    expect(tree.getByTestId("menu")).toBeVisible();
  });
});
