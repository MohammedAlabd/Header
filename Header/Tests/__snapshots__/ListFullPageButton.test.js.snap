// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListFullPageButton Test matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary"
        data-testid="openMenuButton"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          <h6
            class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorPrimary"
          >
            Page
          </h6>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
    <div
      aria-hidden="true"
      class="MuiPopover-root"
      data-testid="menu"
      role="presentation"
      style="position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px; visibility: hidden;"
    >
      <div
        data-test="sentinelStart"
        tabindex="0"
      />
      <div
        class="MuiPaper-root MuiMenu-paper MuiPopover-paper MuiPaper-elevation8 MuiPaper-rounded"
        style="opacity: 0; transform: scale(0.75, 0.5625); visibility: hidden;"
        tabindex="-1"
      >
        <ul
          class="MuiList-root MuiMenu-list MuiList-padding"
          role="menu"
          tabindex="-1"
        >
          <a
            class="makeStyles-textDecoration-1"
            href="/page-url/subPage1-url"
            tabindex="0"
          >
            <li
              aria-disabled="false"
              class="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
              role="menuitem"
              tabindex="-1"
            >
              <h6
                class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorPrimary"
              >
                SUBPAGE1
              </h6>
              <span
                class="MuiTouchRipple-root"
              />
            </li>
          </a>
        </ul>
      </div>
      <div
        data-test="sentinelEnd"
        tabindex="0"
      />
    </div>
  </body>,
  "container": <div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary"
      data-testid="openMenuButton"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        <h6
          class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorPrimary"
        >
          Page
        </h6>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
